lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -654,10 +675,210 @@ namespace Intl
             ret = 0;
         }
 
-LCloseCollator:
         ucol_close(collator);
         return ret;
     }
+
+    // Gets the system time zone. If tz is non-null, it is written into tz
+    // Returns the number of characters written into tz (including a null terminator)
+    int GetDefaultTimeZone(_Out_writes_opt_(tzLen) char16 *tz, _In_ int tzLen)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        ICU_FIXBUF(char16, tz, tzLen); // sets bool isTemporaryBuffer
```

Comment:
>isTemporaryBuffer [](start = 52, length = 17)

trying to understand semantics: isTemporaryBuffer is false if tz is nullptr? So that's effectively controlled by the caller of this function>?

Reply:
Yes -- I considered making different entry points but since ICU is inconsistent about when it accepts null buffers and when it doesnt, I thought I would make my API consistent and fix it when ICU forces me to.